presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -0,0 +1,234 @@
+            this.channels = requireNonNull(channels, "channels is null");
+            verify(channels.stream().map(channel -> channel.filterId).collect(toSet()).size() == channels.size(),
+                    "duplicate dynamic filters are not allowed");
+            verify(channels.stream().map(channel -> channel.index).collect(toSet()).size() == channels.size(),
+                    "duplicate channel indices are not allowed");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            return new DynamicFilterSourceOperator(
+                    driverContext.addOperatorContext(operatorId, planNodeId, DynamicFilterSourceOperator.class.getSimpleName()),
+                    dynamicPredicateConsumer,
+                    channels,
+                    planNodeId);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterSourceOperatorFactory");
+        }
+    }
+
+    private final OperatorContext context;
+    private boolean finished;
+    private Page current;
+    private final Consumer<TupleDomain<String>> dynamicPredicateConsumer;
+    private final int maxFilterPositionsCount;
+    private final long maxFilterSizeInBytes;
+
+    private final List<Channel> channels;
+
+    // May be dropped if the predicate becomes too large.
+    private BlockBuilder[] blockBuilders;
+    private TypedSet[] valueSets;
+
+    private int filterPositionsCount;
+    private long filterSizeInBytes;
+
+    public DynamicFilterSourceOperator(OperatorContext context, Consumer<TupleDomain<String>> dynamicPredicateConsumer, List<Channel> channels, PlanNodeId planNodeId)
+    {
+        this.context = requireNonNull(context, "context is null");
+        this.maxFilterPositionsCount = getDynamicFilteringMaxPerDriverValueCount(context.getSession());
+        this.maxFilterSizeInBytes = getDynamicFilteringMaxPerDriverByteSize(context.getSession()).toBytes();
+
+        this.dynamicPredicateConsumer = requireNonNull(dynamicPredicateConsumer, "dynamicPredicateConsumer is null");
+        this.channels = requireNonNull(channels, "channels is null");
+
+        final int expectedSize = 8;
+        this.blockBuilders = new BlockBuilder[channels.size()];
+        this.valueSets = new TypedSet[channels.size()];
+        for (int channelIndex = 0; channelIndex < channels.size(); ++channelIndex) {
+            Type type = channels.get(channelIndex).type;
+            this.blockBuilders[channelIndex] = type.createBlockBuilder(null, expectedSize);
+            this.valueSets[channelIndex] = new TypedSet(type, Optional.empty(), blockBuilders[channelIndex], expectedSize,
+                                                        String.format("DynamicFilterSourceOperator_%s_%d", planNodeId, channelIndex));
```

Comment:
`TypedSet` has static mem limit. Currently it's `io.prestosql.operator.aggregation.TypedSet#MAX_FUNCTION_MEMORY`.
Either we limit max dynamic filter size or we make `TypedSet` configurable

Reply:
Made `TypedSet` configurable.